fbuapp/View Controllers/RegisterViewController.m

```diff
@@ -0,0 +1,107 @@
+//
+//  RegisterViewController.m
+//  fbuapp
+//
+//  Created by belchercd on 7/22/19.
+//  Copyright © 2019 eazheng. All rights reserved.
+//
+
+#import "RegisterViewController.h"
+#import "AppDelegate.h"
+#import "LogViewController.h"
+#import <FBSDKCoreKit/FBSDKCoreKit.h>
+#import <FBSDKLoginKit/FBSDKLoginKit.h>
+#import "Parse/Parse.h"
+#import "FBSDKProfile.h"
+
+@interface RegisterViewController ()
+
+
+@end
+
+@implementation RegisterViewController
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+
+    // maybe use --> dispatch_async(dispatch_get_main_queue(), ^{
+    if ([FBSDKAccessToken currentAccessToken]) {
+        
+        //FBSDK --> gets user's name and user ID
+        [FBSDKProfile loadCurrentProfileWithCompletion:^(FBSDKProfile *profile, NSError *error) {
+            if (profile){
+                //access to user's email --> error: only getting "email" and not the actual value
+                [[[FBSDKGraphRequest alloc] initWithGraphPath:@"me" parameters:@{@"fields": @"email"}]
+                 startWithCompletionHandler:^(FBSDKGraphRequestConnection *connection, id result, NSError *error) {
+                     if (!error) {
+                         NSLog(@"user:%@", result);
+                     }
+                 }];
+                //creating dictionary to pull out and utilized the user's email
+                NSMutableDictionary* parameters = [NSMutableDictionary dictionary];
+                [parameters setValue:@"email" forKey:@"fields"];
```

Comment:
Here, the value of "fields" is "email" so when you set self.emailField.text = parameters[@"fields"] it basically is doing self.emailField.text = "email"